File: android/src/main/java/com/chessbook/audio_playing/AudioPlayingCheckerPlugin.java
package com.chessbook.audio_playing;

import com.getcapacitor.JSObject;
import com.getcapacitor.Plugin;
import com.getcapacitor.PluginCall;
import com.getcapacitor.PluginMethod;
import com.getcapacitor.annotation.CapacitorPlugin;

@CapacitorPlugin(name = "AudioPlayingChecker")
public class AudioPlayingCheckerPlugin extends Plugin {

    private AudioPlayingChecker implementation = new AudioPlayingChecker();

    @PluginMethod
    public void echo(PluginCall call) {
        String value = call.getString("value");

        JSObject ret = new JSObject();
        ret.put("value", implementation.echo(value));
        call.resolve(ret);
    }

    @PluginMethod
    public void isPlayingAudio(PluginCall call) {
        JSObject ret = new JSObject();
        ret.put("playing", false);
        call.resolve(ret);
    }
}

File: ios/Plugin/AudioPlayingCheckerPlugin.swift
import Foundation
import Capacitor
import AVFoundation

/**
 * Please read the Capacitor iOS Plugin Development Guide
 * here: https://capacitorjs.com/docs/plugins/ios
 */
@objc(AudioPlayingCheckerPlugin)
public class AudioPlayingCheckerPlugin: CAPPlugin {
    private let implementation = AudioPlayingChecker()

    @objc func echo(_ call: CAPPluginCall) {
        let value = call.getString("value") ?? ""
        call.resolve([
            "value": implementation.echo(value)
        ])
    }

    @objc func isPlayingAudio(_ call: CAPPluginCall) {
        call.resolve([
            "playing": AVAudioSession.sharedInstance().secondaryAudioShouldBeSilencedHint
        ])
    }
}
